lib/models/api-key.ts

```diff
@@ -170,6 +171,56 @@ const getApiKeysModel = function (
 			});
 		},
 
+		/**
+		 * @summary Get all provisioning API keys by application (asc)
+		 * @name getProvisioningApiKeysByApplication
+		 * @public
+		 * @function
+		 * @memberof balena.models.apiKey
+		 *
+		 * @param {Object} [options={}] - extra pine options to use
+		 * @fulfil {Object[]} - apiKeys
+		 * @returns {Promise}
+		 *
+		 * @example
+		 * balena.models.apiKey.getProvisioningApiKeysByApplication().then(function(apiKeys) {
+		 * 	console.log(apiKeys);
+		 * });
+		 *
+		 * @example
+		 * balena.models.apiKey.getProvisioningApiKeysByApplication(function(error, apiKeys) {
+		 * 	if (error) throw error;
+		 * 	console.log(apiKeys);
+		 * });
+		 */
+		async getProvisioningApiKeysByApplication(
```

Comment:
Since we say "ByApplication" the first argument should be an "appIdOrSlug: number | string" and then use application.get() in the implementation.
Otherwise this would return all provisioning API keys across all apps.

PS: The method will still have an options argument as it has right now, but as a second param.

Reply:
Right. Thanks for the feedback here. 

I took this as a method which should return keys "ByApplication" as-in all applications in order for a user. Hence the use of `getAll()`.